torch/onnx/symbolic_opset9.py

```diff
@@ -2879,6 +2879,88 @@ def try_mask_to_index(index):
             return sym_help._reshape_helper(g, self, final_shape)
 
 
+@parse_args("v", "v", "is", "i", "v")
+def linalg_norm(g, self, ord, dim, keepdim, dtype):
+    # Conditions based on https://pytorch.org/docs/stable/generated/torch.linalg.norm.html
+    if dim is None:
+        if sym_help._is_none(ord):
+            self = sym_help._reshape_helper(g, self, [-1])
+            ord = g.op("Constant", value_t=torch.LongTensor([2]))
+        self_dim = sym_help._get_tensor_rank(self)
+        if self_dim is None:
+            return _unimplemented("dim",
+                                  "Input rank must be known at export time.")
+        if self_dim == 1:
+            ord_value = sym_help._parse_arg(ord, "f")
+            output = linalg_vector_norm(g, self, ord_value, dim, keepdim, dtype)
+        else:
+            output = linalg_matrix_norm(g, self, ord, [0, 1], keepdim, dtype)
+    else:
+        if len(dim) == 1:
+            if sym_help._is_none(ord):
+                ord = g.op("Constant", value_t=torch.LongTensor([2]))
+            ord_value = sym_help._parse_arg(ord, "f")
+            output = linalg_vector_norm(g, self, ord_value, dim, keepdim, dtype)
```

Comment:
this section is a little repeated from above. Can you organize the code more closely and merge the two?